app/services/requirements/scheduled_datetime_checker.rb

```diff
@@ -2,76 +2,82 @@
-        issues << Issue.new(:scheduled_datetime, :invalid, field: "date")
-      end
+      issues = [check_datetime_is_future,
+                check_datetime_is_foreseeable].compact
+      CheckerIssues.new(issues)
+    end
 
-      if time.blank?
-        issues << Issue.new(:scheduled_datetime, :invalid, field: "time")
-      end
+    def parsed_datetime
+      @parsed_datetime ||= begin
+                             day, month, year = params[:scheduled_date]&.values_at(:day, :month, :year)
+                             Time.zone.strptime(
+                               "#{day}-#{month}-#{year} #{params[:scheduled_time]}",
+                               "#{DATE_FORMAT} #{TIME_FORMAT}",
+                             )
+                           end
+    end
 
-      if issues.any?
-        return CheckerIssues.new(issues)
-      end
+  private
+
+    def check_datetime_valid
+      issues = []
 
       begin
-        if in_the_past?
-          issues << Issue.new(:scheduled_datetime, :in_the_past)
-          return CheckerIssues.new(issues)
-        end
-
-        if too_far_in_the_future?
-          issues << Issue.new(:scheduled_datetime,
-                              :too_far_in_future,
-                              time_period: time_period_for_issue(MAXIMUM_FUTURE_TIME_PERIOD))
-        end
-
-        if too_close_to_now?
-          issues << Issue.new(:scheduled_datetime,
-                              :too_close_to_now,
-                              time_period: time_period_for_issue(MINIMUM_FUTURE_TIME_PERIOD))
-        end
+        day, month, year = params[:scheduled_date]&.values_at(:day, :month, :year)
+        Date.strptime("#{day}-#{month}-#{year}", DATE_FORMAT)
       rescue ArgumentError
-        issues << Issue.new(:scheduled_datetime, :invalid_input)
+        issues << Issue.new(:scheduled_date, :invalid)
       end
 
-      CheckerIssues.new(issues)
-    end
+      begin
+        Time.strptime(params[:scheduled_time].to_s, TIME_FORMAT)
+      rescue ArgumentError
+        issues << Issue.new(:scheduled_time, :invalid)
+      end
 
-    def parsed_datetime
-      @parsed_datetime ||= Time.zone.parse("#{year}-#{month}-#{day} #{time}")
+      issues
     end
 
-  private
+    def check_datetime_is_future
+      if parsed_datetime < Time.current
+        field = parsed_datetime.today? ? :scheduled_time : :scheduled_date
+        return Issue.new(field, :in_the_past)
+      end
 
-    def in_the_past?
-      now = Time.zone.now
-      parsed_datetime <= now
+      if parsed_datetime < Time.current.advance(MINIMUM_FUTURE_TIME_PERIOD)
+        Issue.new(
+          :scheduled_time,
+          :too_close_to_now,
+          time_period: time_period_for_issue(MINIMUM_FUTURE_TIME_PERIOD),
+        )
+      end
     end
 
-    def too_far_in_the_future?
-      now = Time.zone.now
-      parsed_datetime > now.advance(MAXIMUM_FUTURE_TIME_PERIOD).end_of_day
-    end
+    def check_datetime_is_foreseeable
```

Comment:
I don't think foreseeable is the right word here; it's more about predictability and we've got a concrete future time period that we don't need to predict.

Reply:
Sure, any suggestions on the wording? I was hoping I could get something with an is rather than isn't and found isnt_too_far_in_the_future a bit wordy.

somethings considered and rejected: is_realistic is_soonish is_sane

Reply:
`within_time_limit`? I appreciate it still has multiple words but I don't think there is a single word to describe this.